state/execution.go

```diff
@@ -117,12 +116,12 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 	block := state.MakeBlock(height, txs, commit, evidence, proposerAddr)
 
 	localLastCommit := buildLastCommitInfo(block, blockExec.store, state.InitialHeight)
-	rpp, err := blockExec.proxyApp.PrepareProposalSync(
-		abci.RequestPrepareProposal{
+	rpp, err := blockExec.proxyApp.PrepareProposal(context.TODO(),
+		&abci.RequestPrepareProposal{
 			MaxTxBytes:         maxDataBytes,
 			Txs:                block.Txs.ToSliceOfBytes(),
 			LocalLastCommit:    extendedCommitInfo(localLastCommit, votes),
-			Misbehavior:        block.Evidence.Evidence.ToABCI(),
+			Misbehaviors:       block.Evidence.Evidence.ToABCI(),
```

Comment:
Looking at this change, this will break v0.37 implementations. While I think `Misbehaviors` is a better name, I think we should avoid breaking implementations for this.